### PR TITLE
Add OEIS Sequemce A000005 - Count number of divisors

### PR DESCRIPTION
**Pull request by RichardVasquez**
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Sequences/DivisorsCountSequence.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                        BigInteger.DivRem(numerator, divisor, out var remainder);
+                        if (remainder == 0)
+                        {
+                            count++;
+                        }
+                    }
+
+                    n++;
+                    yield return count;
+                }
```
Comment (siriak) [inline Algorithms/Sequences/DivisorsCountSequence.cs:40]: How does it work? Why not as below?
```suggestion
                yield return BigInteger.One;
                for (var n = new BigInteger(2); ; n++)
                {
                    var count = 2;
                    for (var k = 2; k < n; k++)
                    {
                        BigInteger.DivRem(n, k, out var remainder);
                        if (remainder == 0)
                        {
                            count++;
                        }
                    }

                    yield return count;
                }
```

**Review by siriak — CHANGES_REQUESTED** [review]
Hi, looks good, but I'm a bit confused as to how it works. See comment below.

File: Algorithms/Sequences/DivisorsCountSequence.cs (hunk 1 above)
Comment (RichardVasquez) [reply Algorithms/Sequences/DivisorsCountSequence.cs:40]: Hm.  I see your point.  Yours is cleaner.

I went looking through the solutions in OEIS, and the implementation I used was the one that looked best at the end of the evening.  I think I'll sleep on my A000008 before submitting that one.

File: Algorithms.Tests/Sequences/DivisorsCountSequenceTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using NUnit.Framework;
+
+namespace Algorithms.Tests.Sequences
+{
+    public class DivisorsCountSequenceTests
+    {
+        [Test]
+        public void First10ElementsCorrect()
+        {
+            // Values taken from https://oeis.org/A000005 for comparison.
```
Comment (siriak) [inline Algorithms.Tests/Sequences/DivisorsCountSequenceTests.cs:14]:
```suggestion
            // These values are taken from https://oeis.org/A000005 for comparison.
```
Just to restart checks :)

**Comment by siriak** [comment]
Could you take a look at the failing test? It's not from your code, but would be nice to fix.

**Review by siriak — APPROVED** [review]

**Review by siriak — APPROVED** [review]

**Comment by RichardVasquez** [comment]
> Could you take a look at the failing test? It's not from your code, but would be nice to fix.

Work day is starting for me, but I'll take a look at it afterwards.

**Comment by siriak** [comment]
I've already fixed them before merge

**Pull request merged**